### PR TITLE
[TRA-99] Add E2E test for isolated subaccounts having order matches / placing orders.

### PR DESCRIPTION
**Pull request by vincentwschau**
Add E2E test for placing orders on isolated subaccounts, placing orders for isolated markets on subaccounts with cross positions, and isolating / un-isolating accounts by opening / closing positions.

### Changelist
Added E2E tests.

### Test Plan
N/A

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/x/clob/e2e/isolated_subaccount_orders_test.go [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+			},
+			// No changes as no match should have happened.
+			expectedCollateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 10_000_000_000,
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 10_000_000_000,
+			},
+		},
+		"Cross subaccount (with cross position) will not have matches for cross-market orders": {
```
Comment (jayy04) [inline protocol/x/clob/e2e/isolated_subaccount_orders_test.go:183]:
```suggestion
		"Cross subaccount (with cross position) will not have matches for isolated orders": {
```
?

File: protocol/x/clob/e2e/isolated_subaccount_orders_test.go [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 30_000_000_000, // $30,000 USDC
+			},
+			expectedOrdersFilled: []clobtypes.OrderId{},
+			expectedSubaccounts: []satypes.Subaccount{
+				Alice_Num0_MoreIsolatedAfterMatch,
+				Bob_Num0_CrossAfterMatch,
+			},
+			// No changes as no match should have happened.
```
Comment (jayy04) [inline protocol/x/clob/e2e/isolated_subaccount_orders_test.go:295]:
```suggestion
```

File: protocol/x/clob/e2e/isolated_subaccount_orders_test.go [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+				// $10,000 USDC + $10,000 USDC + (match) 100 quote quantums - fee (1 quote quantum)
+				satypes.ModuleAddress.String(): 20_000_000_099,
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+					// $30,000 USDC - $10,000 USDC - (match) 100 quote quantums
+				).String(): 19_999_999_900,
+			},
+		},
+		`Empty subaccount will not have matches for isolated market orders if cross collateral pool does not 
+		have enough balance`: {
```
Comment (jayy04) [inline protocol/x/clob/e2e/isolated_subaccount_orders_test.go:305]: for my understanding - this is something that can potentially happen? why would cross collat pool not have enough balance?

Comment (jayy04) [reply protocol/x/clob/e2e/isolated_subaccount_orders_test.go:305]: attacks from -ve TNC / bypassing withdrawal gating?

File: protocol/x/clob/e2e/isolated_subaccount_orders_test.go [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+			expectedCollateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 1_000_000_000, // $1,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 30_000_000_000, // $30,000 USDC
+			},
+			expectedErr: "insufficient funds",
+		},
+		`Isolated subaccount will not have matches to close isolated perpetual position if isolated collateral pool does not 
+		have enough balance`: {
```
Comment (jayy04) [inline protocol/x/clob/e2e/isolated_subaccount_orders_test.go:345]: same here - when would this happen? when withdrawal gating fails?

**Review by jayy04 — APPROVED** [review]

File: protocol/x/clob/e2e/isolated_subaccount_orders_test.go (hunk 3 above)
Comment (vincentwschau) [reply protocol/x/clob/e2e/isolated_subaccount_orders_test.go:305]: Yes, exactly.

File: protocol/x/clob/e2e/isolated_subaccount_orders_test.go (hunk 4 above)
Comment (vincentwschau) [reply protocol/x/clob/e2e/isolated_subaccount_orders_test.go:345]: Yep, this is a problem as well.

**Pull request merged**